lib/kredis/types/expirable.rb

```diff
@@ -0,0 +1,42 @@
+module Kredis::Types::Expirable
+  extend ActiveSupport::Concern
+
+  included do
+    proxying :expire, :expireat, :ttl
+
+    attr_accessor :expires_in, :expires_at
+
+    def expire_in(seconds)
+      expire seconds.to_i
+    end
+
+    def expire_at(datetime)
+      expireat datetime.to_i
+    end
+
+    def refresh_expiration
+      expire_in(@expires_in) if expires_in
+      expire_at(@expires_at) if expires_at && !expires_in
+    end
+  end
+
+  def self.on(*on_methods)
+    Module.new do
+      define_singleton_method :included do |type_klass|
+        type_klass.include Kredis::Types::Expirable
+
+        type_klass.prepend(Module.new do
+          on_methods.each do |method|
+            define_method method do |*args, **kargs, &block|
+              super(*args, **kargs, &block)
```

Comment:
Notice this would change the return method. You could use `tap`:

```ruby
super(*args, **kargs, &block).tap do
  refresh_expiration
end
```

I initially thought that there should be a transaction wrapping the original code and the `refresh_expiration logic`. But then I thought that it's probably not worth the hassle. I think it's acceptable that these `expire` operations aren't executed transactionally with the intercepted operation. In case of a race condition, the last refresh would prevail, and that's fine. If we wanted to do it, the problem is that Redis doesn't support nested `multi` operations. If we got #78 merged, maybe we could use that to detect if there is an ongoing transaction or we need to start one (assuming there is an expiration date to use). But as mentioned, I don't think it's worth the trouble.

Reply:
There is also an option when sending the EXPIRE command to Redis to only update if the new value is greater than the existing TTL. That could be helpful for the race-condition case.

Reply:
> Notice this would change the return method. You could use tap:

Not that easy, [`Redis::Future` inherits from `BasicObject`](https://www.rubydoc.info/github/redis/redis-rb/Redis/Future).